util/parsenum.h

```diff
@@ -15,8 +15,14 @@
  * nonzero with an unspecified value of ${x} and errno set to EINVAL or ERANGE
  * as appropriate.
  */
+#ifdef __clang__
+/* Disable clang warnings arising from implicit conversion in PARSENUM. */
 #define PARSENUM(x, s, min, max)					\
 	(								\
+_Pragma("clang diagnostic push")					\
```

Comment:
Ooh, I didn't realize `_Pragma` was C99.  OK, let's do it this way, but rather than having two definitions for PARSENUM, define PARSENUM_PROLOGUE and PARSENUM_EPILOGUE as being either these pragmas or nothing.